lib/msf/core/exe/segment_appender.rb

```diff
@@ -32,7 +32,12 @@ def generate_pe
 
       # Create a new section
       s = Metasm::PE::Section.new
-      s.name = '.' + Rex::Text.rand_text_alpha_lower(4)
+      if secname.blank?
+        s.name = '.' + Rex::Text.rand_text_alpha_lower(4)
+      else
+        s.name = '.' + secname.downcase
+        $stderr.puts "Created custom section \"#{s.name}\""
```

Comment:
Library code shouldn't log to $stderr directly

Reply:
Thanks. I'm considering two things:
1. Trashing the log statement entirely
2. Logging using cli_print within `payload_generator.rb` as such:
`def exe_options`
`  ...`
`  unless secname.blank?`
`    opts[:secname]       = secname`
`    cli_print "Creating custom section \"#{secname}\""`
`  end`
which is less assuring because it prints the input section name, not the literal section name. `s.name`, hence my use of "Creating" rather than "Created"

Any advice on this?